### DISCUSSION TITLE
what is the rationale behind `var i: i32` ,vs. `var i  i32`

### DISCUSSION THREAD

**Discussion by gintominto5329** [carbon-language/carbon-lang#3603] [Language design]
thanks

**Comment by chandlerc** [comment]
The details of this change are linked from design document describing variables.

The proposal that set the ordering also included adding the `:`: https://github.com/carbon-language/carbon-lang/blob/trunk/proposals/p0618.md

The syntax design and proposal doesn't discuss `var <name> <type>` (without the colon) because how few other languages use that syntax (Go is the only notable one I'm aware of), and how many languages use the form with the colon. A major component for the rationale was matching what has become widespread among modern and popular languages which is overwhelmingly dominated by syntaxes with the `:`.
